src/language-js/printer-estree.js

```diff
@@ -3870,6 +3870,23 @@ function printArgumentsList(path, options, print) {
     ]);
   }
 
+  // useEffect(() => { ... }, [foo, bar, baz])
+  if (
+    args.length === 2 &&
```

Comment:
I'm a bit late with this but, this misses useImperativeHandle which also takes a deps array and has a length of 3 when the array is present.

I'm not sure if it'd be that much better to support it, though; useImperativeHandle specifically has a very short first argument (it's just `ref,` almost always) which wouldn't look too bad; but what if it catches other custom hooks?

Maybe a heuristic could be that, if the length is 3 and the first argument is very short (3 characters or less) then keep it in one line, otherwise split?